lib/asciidoctor.rb

```diff
@@ -7,10 +7,9 @@
 
 require 'set'
 
-# NOTE RUBY_ENGINE == 'opal' conditional blocks are filtered by the Opal preprocessor
+# NOTE RUBY_ENGINE == 'opal' conditional blocks like this are filtered by the Opal preprocessor
 if RUBY_ENGINE == 'opal'
-  # NOTE asciidoctor/opal_ext is supplied by the Asciidoctor.js build
-  require 'asciidoctor/opal_ext'
+  require 'asciidoctor/opal' # supplies compile and runtime overrides for the Opal environment; fulfilled by Asciidoctor.js
```

Comment:
I'm happy with a module named "js". It would essentially replace "opal". So the includes would be:

```ruby
require 'asciidoctor/js' # at top of asciidoctor.rb
```

and

```ruby
require 'asciidoctor/js/postscript' # at bottom of asciidoctor.rb
```

Here's what `asciidoctor/js` would look like:

```ruby
require 'asciidoctor/js/opal_ext'
require 'asciidoctor/js/runtime'
require 'strscan'
require 'asciidoctor/js/rx'
```

Here's what `asciidoctor/js/postscript` would look like:

```ruby
require 'asciidoctor/converter/composite'
require 'asciidoctor/converter/html5'
require 'asciidoctor/extensions'
```

With this layout, the regular expression constants will be defined before anything else in the Asciidoctor module is defined. But I think this is completely reasonable.